scripts/rpm/post-remove.sh

```diff
@@ -1,55 +1,18 @@
 #!/bin/bash
 
-function disable_systemd {
-    systemctl disable telegraf
-    rm -f $1
-}
+# Telegraf is no longer installed, remove from systemd
+if [[ "$1" = "0" ]]; then
+    rm -f /etc/default/telegraf
 
-function disable_update_rcd {
-    update-rc.d -f telegraf remove
-    rm -f /etc/init.d/telegraf
-}
-
-function disable_chkconfig {
-    chkconfig --del telegraf
-    rm -f /etc/init.d/telegraf
-}
-
-if [[ -f /etc/redhat-release ]] || [[ -f /etc/SuSE-release ]]; then
-    # RHEL-variant logic
-    if [[ "$1" = "0" ]]; then
-        # InfluxDB is no longer installed, remove from init system
-        rm -f /etc/default/telegraf
-
-        if [[ "$(readlink /proc/1/exe)" == */systemd ]]; then
-            disable_systemd /usr/lib/systemd/system/telegraf.service
-        else
-            # Assuming sysv
-            disable_chkconfig
-        fi
+    if [[ -d /run/systemd/system ]]; then
+        systemctl disable telegraf
+        rm -f /usr/lib/systemd/system/telegraf.service
```

Comment:
Do we need a `systemctl daemon-reload` here too after the file has been removed?

Reply:
yes! good catch